src/RenderWebGL.js

```diff
@@ -728,23 +741,23 @@ class RenderWebGL extends EventEmitter {
         const color = __touchingColor;
         const hasMask = Boolean(mask3b);
 
-        for (let y = bounds.bottom; y <= bounds.top; y++) {
-            if (bounds.width * (y - bounds.bottom) * (candidates.length + 1) >= __cpuTouchingColorPixelCount) {
-                return this._isTouchingColorGpuFin(bounds, color3b, y - bounds.bottom);
+        // Scratch Space - +y is top
+        for (let y = 0; y < bounds.height; ++y) {
+            if (bounds.width * y * (candidates.length + 1) >= __cpuTouchingColorPixelCount) {
+                return this._isTouchingColorGpuFin(bounds, color3b, y);
             }
-            // Scratch Space - +y is top
-            for (let x = bounds.left; x <= bounds.right; x++) {
-                point[1] = y;
-                point[0] = x;
-                if (
-                    // if we use a mask, check our sample color
-                    (hasMask ?
-                        maskMatches(Drawable.sampleColor4b(point, drawable, color), mask3b) :
-                        drawable.isTouching(point)) &&
-                    // and the target color is drawn at this pixel
-                    colorMatches(RenderWebGL.sampleColor3b(point, candidates, color), color3b, 0)
-                ) {
-                    return true;
+            for (let x = 0; x < bounds.width; ++x) {
+                point[1] = bounds.top - y; // bounds.top <= y < bounds.bottom ("flipped")
```

Comment:
I really don't like the fact that this ends up with `y !== point[1]` -- can we maybe rename the outer loop var to `yFromTop` or something

Reply:
Hmm, I kind of agree but on the other hand if you combine this with #406 then you end up with `debugCanvasContext.fillRect(x, y, 1, 1)` being the correct way to draw to the debug canvas to match the GPU path's results. I'm not saying that clean debug code should be the priority (ha!) but rather I feel like that's a sign that these are the "natural" values for `x` and `y`. (Side note: it's very satisfying to see the CPU path early-out when a matching color is found, leaving the rest of the canvas blank!)

Consider also that `0 <= y < bounds.height` and `0 <= x < bounds.width` are both correct here, as is `bounds.left <= point[0] < bounds.right`, but `bounds.bottom <= point[1] < bounds.top` is wrong.

Actually, that makes me realize that this comment refers to the wrong variable. I'll fix that :/